src/test/java/seedu/address/model/person/HoursWorkedTest.java

```diff
@@ -15,46 +16,119 @@ public void constructor_null_throwsNullPointerException() {
         String floatingPointHours = "3.5";
         assertThrows(IllegalArgumentException.class,
-                LeaveBalance.MESSAGE_CONSTRAINTS, () -> new LeaveBalance(floatingPointHours));
+                HoursWorked.MESSAGE_CONSTRAINTS, () -> new HoursWorked(floatingPointHours));
     }
 
     @Test
     public void isValidHoursWorked() {
         // null hoursWorked
         assertThrows(NullPointerException.class, () -> HoursWorked.isValidHoursWorked(null));
 
-        // invalid hoursWorked
+        // invalid hoursWorked (formatting)
         assertFalse(HoursWorked.isValidHoursWorked("")); // empty string
         assertFalse(HoursWorked.isValidHoursWorked(" ")); // spaces only
-        assertFalse(HoursWorked.isValidHoursWorked("seven days")); // contains non-numeric characters only
+        assertFalse(HoursWorked.isValidHoursWorked("seven")); // contains non-numeric characters only
         assertFalse(HoursWorked.isValidHoursWorked("7h")); // contains alphanumeric characters
-        assertFalse(HoursWorked.isValidHoursWorked("-1")); // contains negative values
+        assertFalse(HoursWorked.isValidHoursWorked("-0")); // contains negative 0
         assertFalse(HoursWorked.isValidHoursWorked("1.1")); // contains floating point values
-        assertFalse(HoursWorked.isValidHoursWorked("12345")); // exceeds max bound
 
-        // valid hoursWorked
-        assertTrue(HoursWorked.isValidHoursWorked("365")); // max boundary
+        // invalid hoursWorked (equivalence partitions: [INTEGER_MIN, -1], [745, INTEGER_MAX])
+        assertFalse(HoursWorked.isValidHoursWorked("745")); // boundary value
+        assertFalse(HoursWorked.isValidHoursWorked("-1")); // boundary value
+        assertFalse(HoursWorked.isValidHoursWorked("12345")); // value inside equivalence partition
+        assertFalse(HoursWorked.isValidHoursWorked("-12345")); // value inside equivalence partition
+
+        // valid hoursWorked (equivalence partition: [0, 744])
+        assertTrue(HoursWorked.isValidHoursWorked("744")); // max boundary
         assertTrue(HoursWorked.isValidHoursWorked("0")); // min boundary
-        assertTrue(HoursWorked.isValidHoursWorked("100")); // value inside equivalence partition
-        assertTrue(HoursWorked.isValidHoursWorked("201")); // value inside equivalence partition
+        assertTrue(HoursWorked.isValidHoursWorked("087")); // value inside equivalence partition
+        assertTrue(HoursWorked.isValidHoursWorked("367")); // value inside equivalence partition
+    }
+
+    @Test
+    public void addHoursWorked_throwsAssertionError() {
+        HoursWorked validHoursWorked = new HoursWorked("10");
+        HoursWorked noHoursWorked = new HoursWorked("0"); // Negative values are caught by isValidHoursWorked
+        assertThrows(AssertionError.class, () -> validHoursWorked.addHoursWorked(noHoursWorked));
+    }
+
+    @Test
+    public void addHoursWorked_throwsIllegalArgumentException() {
```

Comment:
Maybe you might want to add a comment that this fails because total exceeds the upper bound for HoursWorked?